ros/angel_system_nodes/angel_system_nodes/gpt_intent_detector.py

```diff
@@ -0,0 +1,100 @@
+from langchain import PromptTemplate, FewShotPromptTemplate
+from langchain.chains import LLMChain
+from langchain.chat_models import ChatOpenAI
+from langchain.llms import OpenAI
+import openai
+import os
+import rclpy
+
+from angel_system_nodes.base_intent_detector import BaseIntentDetector, INTENT_LABELS
+
+openai.organization = os.getenv("OPENAI_ORG_ID")
+openai.api_key = os.getenv("OPENAI_API_KEY")
+
+# The following are few shot examples when prompting GPT.
+FEW_SHOT_EXAMPLES = [
+    {"utterance": "Go back to the previous step!", "label": "prev_step."},
+    {"utterance": "Next step, please.", "label": "next_step"},
+    {"utterance": "How should I wrap this tourniquet?", "label": "inquiry"},
+    {"utterance": "The sky is blue", "label": "other"},
+]
+
+
+class GptIntentDetector(BaseIntentDetector):
```

Comment:
Are we planning to use this for the November demo?

Reply:
We can  use either the BaseIntentDetector (keyphrase matching) or  GptIntentDetector. I decoupled them such  that we can easily replace one with the other based on our use cases.